src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,499 @@
+    const reportParticipants = lodashGet(props, 'report.participants', []);
+    const participantsWithDetails = _.map(OptionsListUtils.getPersonalDetailsForLogins(reportParticipants, props.personalDetails), personalDetails => ({
+        login: personalDetails.login,
+        text: personalDetails.displayName,
+        firstName: lodashGet(personalDetails, 'firstName', ''),
+        lastName: lodashGet(personalDetails, 'lastName', ''),
+        alternateText: Str.isSMSLogin(personalDetails.login) ? Str.removeSMSDomain(personalDetails.login) : personalDetails.login,
+        icons: [{
+            source: ReportUtils.getAvatar(personalDetails.avatar, personalDetails.login),
+            name: personalDetails.login,
+            type: CONST.ICON_TYPE_AVATAR,
+        }],
+        keyForList: personalDetails.login,
+        payPalMeAddress: lodashGet(personalDetails, 'payPalMeAddress', ''),
+        phoneNumber: lodashGet(personalDetails, 'phoneNumber', ''),
+    }));
+
+    // Skip IOUParticipants step if participants are passed in
+    const steps = reportParticipants.length ? [Steps.IOUAmount, Steps.IOUConfirm] : [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm];
+
+    const prevCreatingIOUTransactionStatusRef = useRef();
+    const prevNetworkStatusRef = useRef();
+    const prevSelectedCurrencyCodeRef = useRef();
+
+    const [previousStepIndex, setPreviousStepIndex] = useState(0);
+    const [currentStepIndex, setCurrentStepIndex] = useState(0);
+    const [participants, setParticipants] = useState(participantsWithDetails);
+    const [amount, setAmount] = useState('');
+    const [comment, setComment] = useState('');
+
+    useEffect(() => {
+        PersonalDetails.openMoneyRequestModalPage();
+        IOU.setIOUSelectedCurrency(props.currentUserPersonalDetails.localCurrencyCode);
+    }, []);
+
+    useEffect(() => {
+        prevCreatingIOUTransactionStatusRef.current = lodashGet(props, 'iou.creatingIOUTransaction');
+    }, [props.iou]);
+
+    useEffect(() => {
+        prevNetworkStatusRef.current = props.network.isOffline;
+    }, [props.network.isOffline]);
+
+    useEffect(() => {
+        prevSelectedCurrencyCodeRef.current = lodashGet(props, 'iou.selectedCurrencyCode');
+    }, [props.iou.selectedCurrencyCode]);
+
+    useEffect(() => {
+        const isDoneCreatingIOUTransaction = prevCreatingIOUTransactionStatusRef.current && !lodashGet(props, 'iou.creatingIOUTransaction');
+
+        // User came back online, so we can try to create the IOU transaction again
+        if (prevNetworkStatusRef.current && !props.network.isOffline) {
+            PersonalDetails.openMoneyRequestModalPage();
+        }
+
+        // If the curency code is updated, also update the IOU selected currency
+        const selectedCurrencyCode = lodashGet(props, 'iou.selectedCurrencyCode');
+        if (prevSelectedCurrencyCodeRef.current !== selectedCurrencyCode) {
+            IOU.setIOUSelectedCurrency(selectedCurrencyCode);
+        }
```

Comment:
NAB, but feel like I don't understand it or maybe it's an anti-pattern. Why is the currency changing? Can we just update this whenever the currency changes (e.g. by moving `IOU.setIOUSelectedCurrency()` closer to the logic that updates the prop instead of applying a side effect)?

Reply:
Yeah I also agree this is an anti-pattern, I would imagine we call this where the logic happens of the user changing their currency code

Reply:
... I went to the spot where the currency is selected and it already does this so I am going to 🔪 it from here and see what breaks